iotile_ext_cloud/iotile/cloud/apps/ota_updater.py

```diff
@@ -0,0 +1,210 @@
+import logging
+import urllib.request
+from datetime import datetime
+import operator
+
+from iotile.core.hw import IOTileApp
+from iotile.core.dev.semver import SemanticVersion
+from iotile.cloud import IOTileCloud, device_id_to_slug
+from typedargs.annotate import docannotate, context
+
+import pytest
+
+op_map = {">=": operator.lt,
+           "gteq": operator.lt,
+           "<=": operator.gt,
+           "lteq": operator.gt,
+           ">": operator.le,
+           "gt": operator.le,
+           "<": operator.ge,
+           "lt": operator.ge,
+           "==": operator.ne,
+           "eq": operator.ne
+          }
+
+
+def _download_ota_script(script_url):
```

Comment:
I would recommend not downloading to disk but just fetching in memory and then passing to `run_script` in `UpdaterApp` which directly takes an `UpdateScript` object:
https://github.com/iotile/coretools/blob/master/iotilecore/iotile/core/hw/app/updater_app.py#L139

The `requests` package has a good http `get` routine that will just give you the binary data in memory.